File: frontend_app/lib/features/chat/model/chat_message.dart
class ChatMessage {
  final String id;
  final String trendId;
  final String text;
  final String senderName;
  final DateTime timestamp;
  final bool isMe;

  ChatMessage({
    required this.id,
    required this.trendId,
    required this.text,
    required this.senderName,
    required this.timestamp,
    required this.isMe, // This remains frontend-only to control UI
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json, String currentUserName) {
    return ChatMessage(
      id: json['_id'] ?? DateTime.now().toString(),
      trendId: json['trendId'] ?? '',
      text: json['text'] ?? '',
      senderName: json['senderName'] ?? 'Unknown',
      timestamp: json['timestamp'] != null 
          ? DateTime.parse(json['timestamp']) 
          : DateTime.now(),
      isMe: json['senderName'] == currentUserName,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'trendId': trendId,
      'text': text,
      'senderName': senderName,
    };
  }
}

File: backend/src/models/ChatMessage.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IChatMessage extends Document {
    trendId: string;
    text: string;
    senderName: string;
    timestamp: Date;
}

const ChatMessageSchema: Schema = new Schema({
    trendId: {
        type: String,
        required: true,
        index: true,
    },
    text: {
        type: String,
        required: true,
        trim: true,
        maxlength: 1000,
    },
    senderName: {
        type: String,
        required: true,
        trim: true,
        maxlength: 50,
    },
    timestamp: {
        type: Date,
        default: Date.now,
        index: true,
    },
});

export default mongoose.model<IChatMessage>('ChatMessage', ChatMessageSchema);
